server/base/src/main/java/org/apache/accumulo/server/conf/CheckCompactionConfig.java

```diff
@@ -0,0 +1,180 @@
+      System.err.println("No properties file was given");
+      System.exit(1);
+    }
+
+    Path path = Path.of(opts.filePath);
+    if (!path.toFile().exists()) {
+      System.err.println("File at given path could not be found");
+      System.exit(1);
+    }
+
+    // Extract properties from props file at given path
+    Properties allProps = ClientInfoImpl.toProperties(path);
+    log.debug("All props: {}", allProps);
+
+    // Extract server props from set of all props
+    Map<String,String> serverPropsMap = new HashMap<>();
+    String suffix = ".accumulo.server.props";
+    allProps.forEach((k, v) -> {
+      if (k.toString().endsWith(suffix))
+        serverPropsMap.put(k.toString(), v.toString());
+    });
+
+    // Ensure there is exactly one server prop in the map and get its value
+    // The value should be compaction properties
+    String compactionPropertiesString = "";
+    try {
+      compactionPropertiesString = Iterables.getOnlyElement(serverPropsMap.values());
+    } catch (IllegalArgumentException e) {
+      System.err.println("There should be exactly one server property. " + e.getMessage());
+      System.exit(1);
+    }
+
+    // Create a props object for compaction props
+    StringReader sr = new StringReader(compactionPropertiesString.replace(' ', '\n'));
+    Properties serverProps = new Properties();
+    serverProps.load(sr);
+
+    // Extract executors options from compactions props
+    Map<String,String> executorsProperties =
+        getPropertiesWithSuffix(serverProps, ".planner.opts.executors");
+
+    // Ensure there is exactly one executor config in the map and get its value
+    String executorJson = "";
+    try {
+      executorJson = Iterables.getOnlyElement(executorsProperties.values());
+    } catch (IllegalArgumentException e) {
+      System.err.println("There should be exactly one executor config. " + e.getMessage());
+      System.exit(1);
+    }
+
+    final String finalExecutorJson = executorJson;
+    CompactionPlanner.InitParameters params = new CompactionPlanner.InitParameters() {
+      @Override
+      public ServiceEnvironment getServiceEnvironment() {
+        return null;
+      }
+
+      @Override
+      public Map<String,String> getOptions() {
+        return Map.of("executors", finalExecutorJson);
+      }
+
+      @Override
+      public String getFullyQualifiedOption(String key) {
+        return null;
+      }
+
+      @Override
+      public ExecutorManager getExecutorManager() {
+        return new ExecutorManager() {
+          @Override
+          public CompactionExecutorId createExecutor(String name, int threads) {
+            return CompactionExecutorIdImpl.externalId(name);
+          }
+
+          @Override
+          public CompactionExecutorId getExternalExecutor(String name) {
+            return CompactionExecutorIdImpl.externalId(name);
+          }
+        };
+      }
+    };
+
+    try {
+      new DefaultCompactionPlanner().parseExecutors(params);
```

Comment:
Did you consider using the planner property to create the planner class instead of assuming its the DefaultPlanner?

Reply:
I hadn't, but now that you mention it that does make more sense. I'll look into that.

Reply:
I am having trouble implementing this because I would need to call `init()` rather than `parseExecutors()` (a method that I created to extract and separate the logic in `init`). This is an issue because `init` relies on objects that will not exist while using this tool. For example, if calling `init()` for DefaultCompactionPlanner, which in turn calls `determineMaxFilesToCompact()`, the params object that is passed in would need to have reference to a `ServiceEnvironment` object. This maybe be an issue for other implementations of `CompactionPlanner` as well since we do not know what will be required within the passed in `CompactionPlanner.InitParameters`.

I'm still looking into whether there is a workaround for this.

Reply:
@DomGarguilo I attempted to refactor some existing code so that it could be used by this utility.  I did that in this [commit](https://github.com/keith-turner/accumulo/commit/b17492345b777b5a8fc267c64b3e02b2333b472b) that is in this [branch](https://github.com/keith-turner/accumulo/commits/DomGarguilo-checkCompactionConfig).

Its not complete and I am not completely sure it will work, but I suspect it will.  Feel free to use it and build on it if you think its workable.

Reply:
Thanks @keith-turner! I think what you have looks good. I'm working through a few details but I think its going to work.

Reply:
> @DomGarguilo I attempted to refactor some existing code so that it could be used by this utility. I did that in this [commit](https://github.com/keith-turner/accumulo/commit/b17492345b777b5a8fc267c64b3e02b2333b472b) that is in this [branch](https://github.com/keith-turner/accumulo/commits/DomGarguilo-checkCompactionConfig).

The current changes the changes you made.